### ISSUE TITLE
Creating AKS Cluster fails

### ISSUE THREAD

**Issue by n0daft** [rancher/rancher#13201]
**Rancher versions:**
rancher/server or rancher/rancher: v2.0.0-rc3
rancher/agent or rancher/rancher-agent:

**Infrastructure Stack versions:**
healthcheck:
ipsec:
network-services:
scheduler:
kubernetes (if applicable):

**Docker version: (`docker version`,`docker info` preferred)**
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 2
Server Version: 17.03.2-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 17
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.13.0-1013-gcp
Operating System: Ubuntu 16.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 4.591 GiB
Name: rancher-server
ID: U3F6:CJN7:W2JD:BK72:CGNM:OUE3:X5WB:K5BK:5IXK:UBK6:AVTV:4O5S
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
WARNING: No swap limit support
**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
4.13.0-1013-gcp
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Rancher Server: GCE
Cluster: AKS
**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher, internal DB
**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Kubernetes
**Steps to Reproduce:**
Add Cluster via Azure Container Service wizard.
**Results:**
Stuck in provisioning. rancher server container logs:
```
2018/04/30 09:37:38 [INFO] Provisioning cluster [c-hqtgj]
2018/04/30 09:37:38 [INFO] Creating cluster [c-hqtgj]
E0430 09:37:38.498742       1 generic_controller.go:204] ClusterController c-hqtgj recovered from panic "runtime error: invalid memory address or nil pointer dereference". (err=<nil>) Call stack:
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:153
/usr/local/go/src/runtime/asm_amd64.s:509
/usr/local/go/src/runtime/panic.go:491
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:367
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/kontainer-engine/types/rpc_client.go:39
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/kontainer-engine/types/rpc_client.go:38
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/kontainer-engine/cluster/cluster.go:151
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/kontainer-engine/cluster/cluster.go:79
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/kontainer-engine/service/service.go:201
/go/src/github.com/rancher/rancher/pkg/controllers/management/clusterprovisioner/driver.go:22
/go/src/github.com/rancher/rancher/pkg/controllers/management/clusterprovisioner/provisioner.go:225
/go/src/github.com/rancher/rancher/pkg/controllers/management/clusterprovisioner/provisioner.go:105
/go/src/github.com/rancher/rancher/pkg/controllers/management/clusterprovisioner/provisioner.go:147
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/condition/condition.go:149
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/condition/condition.go:132
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/condition/condition.go:123
/go/src/github.com/rancher/rancher/pkg/controllers/management/clusterprovisioner/provisioner.go:146
/go/src/github.com/rancher/rancher/pkg/controllers/management/clusterprovisioner/provisioner.go:142
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/types/apis/management.cattle.io/v3/zz_generated_cluster_lifecycle_adapter.go:19
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/lifecycle/object.go:158
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/lifecycle/object.go:57
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/lifecycle/object.go:40
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/types/apis/management.cattle.io/v3/zz_generated_cluster_lifecycle_adapter.go:49
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/types/apis/management.cattle.io/v3/zz_generated_cluster_controller.go:122
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:250
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:190
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:178
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:170
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2337
```

Maybe an authorization failure causes the Nullpointer.

**Comment by n0daft** [comment]
So I had three problems:

1. Wrong tenant-id.
2. Missed assigning role to service principal. I assigned now the owner role, although contributor would probably be enough

After that I noticed that the resource group was created but nothing else. This is at least a sign that authorization is successful. If so check the activity log of the resource group for errors:

3. In my case there was the error

```
The VM size of Agent is not allowed in your subscription in location 'westeurope'. Agent VM size 'Standard_D11' is available in locations: 
australiasoutheast,brazilsouth,centralus,centraluseuap,eastasia,eastus,eastus2,japanwest,northcentralus,southeastasia,westus.
```

So I chose the 'Standard_D11_v2' and now it seems to be working (cluster is spinning up right now).
In any case, there was no feedback at all in the Rancher Web UI.
